src/Bicep.Core/Semantics/AuxiliaryFileCache.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Utils;
+using System.Diagnostics;
+using System.Linq;
+using Bicep.Core.Extensions;
+
+namespace Bicep.Core.Semantics;
+
+public record AuxiliaryFile(
+    Uri FileUri,
+    byte[] Content);
```

Comment:
did you consider using BinaryData instead of byte[]? It has some easy to use methods to easily transform into String and also deserialize into objects